Excel.TemplateEngine/ObjectPrinting/LazyParse/LazyClassParser.cs

```diff
@@ -0,0 +1,128 @@
+                                continue;
+
+                            var templateEnumerableRow = templateRow.SkipWhile(x => x.CellPosition.CellReference != templateCell.CellPosition.CellReference)
+                                                                   .ToArray();
+                            ParseEnumerable(tableReader, model, templateEnumerableRow, enumerableType);
+                            break;
+                        }
+
+                        ParseSingleValue(targetCell, model, path);
+                    }
+                }
+            }
+
+            return model;
+        }
+
+        /// <summary>
+        ///     Read rows one-by-one. Parse only first met enumerable.
+        /// </summary>
+        private void ParseEnumerable([NotNull] LazyTableReader tableReader,
+                                     [NotNull] object model,
+                                     [NotNull] [ItemNotNull] ICell[] templateEnumerableRow,
+                                     [NotNull] Type enumerableType)
+        {
+            var firstExpression = templateEnumerableRow.First().StringValue;
+            var arrayAccessEnd = firstExpression.IndexOf(']');
+            var enumerableExpression = firstExpression.Substring(0, arrayAccessEnd + 1);
+
+            var firstEnumerableTemplateCells = templateEnumerableRow.Where(x => !string.IsNullOrEmpty(x.StringValue) &&
+                                                                                x.StringValue.StartsWith(enumerableExpression))
+                                                                    .ToArray();
+
+            var modelType = model.GetType();
+            var pathToEnumerable = ExcelTemplatePath.FromRawExpression(enumerableExpression);
+            var itemType = ObjectPropertiesExtractor.ExtractChildObjectTypeFromPath(modelType, pathToEnumerable);
+
+            var items = ListParser.Parse(tableReader, modelType, itemType, firstEnumerableTemplateCells, logger);
+
+            var withoutArrayAccess = pathToEnumerable.WithoutArrayAccess();
+            var enumerableSetter = ObjectChildSetterFabric.GetEnumerableSetter(modelType, withoutArrayAccess, enumerableType, itemType);
+
+            enumerableSetter(model, items);
+        }
+
+        private void ParseSingleValue([NotNull] SimpleCell cell,
+                                      [NotNull] object model,
+                                      [NotNull] ExcelTemplatePath leafPath)
+        {
+            var leafSetter = ObjectChildSetterFabric.GetChildObjectSetter(model.GetType(), leafPath);
+            var leafModelType = ObjectPropertiesExtractor.ExtractChildObjectTypeFromPath(model.GetType(), leafPath);
+
+            if (!TextValueParser.TryParse(cell.CellValue, leafModelType, out var parsedObject))
+            {
+                logger.Warn($"Failed to parse value {cell.CellValue} from {cell.CellPosition.CellReference} with type='{leafModelType}'");
```

Comment:
лучше структурированные логи писать https://vostok.gitbook.io/logging/concepts/syntax/passing-properties:
```suggestion
                logger.Warn("Failed to parse value {CellValue} from {CellReference} with type='{Type}'", new {CellValue = call.CellValue, CellReference = cell.CellPosition.CellReference, Type = leafModelType});
```

Reply:
Думаешь, придется искать по CellValue, CellReference?

Reply:
скорее иногда хочется поискать все подобные записи, а по полному совпадению `messageTemplate` фильтровать в кибане как-то проще и быстрее, чем по каким-то паттернам